docs/proposals/20220105-observatorium-alerting.md

```diff
@@ -0,0 +1,169 @@
+
+### Pitfalls of the current solution
+
+Observatorium does not have alerting support.
+
+## Goals
+
+Definition: 
+* Alert Routing: Configuration of Alertmanager related to receiver target config and its routing. 
+
+Goals:
+
+* Tenants should be able to read the status of active alerts 
+  * via ([Prometheus Alert API GET](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts))
+* Tenants are able to configure alert routing:
+  * via new GET, DELETE and PUT HTTP API
+  * Potentially work with Alertmanager Team on upstream API.
+* Both alerts and alerting rules should be multi-tenant.
+
+## Non-Goals
+
+* Tenant should be able to create and edit alerting rules:
+  * Alerting Rules via GET and PUT via HTTP (Rules API)
+  * This is not the goal, because it should be done when recording rule API is done.
+* UI for Alerts/Rules, Alert Status, AM Routing, AM Alerts …
+* Alertmanager Alerts API, Silences, Inhibits API
+  * We should focus on that in next iteration.
+* Scaling Rulers
+* Different configuration than Alertmanager with 2 replicas.
+* Testing routing configuration or UI allowing better experience for configuring those. It is a common issue that routing can be very misleading. 
+
+## Abbreviations
+
+AM: Alertmanager
+
+## Audience of this proposal
+
+Observatorium Devs and Users.
+
+## How
+
+Within work for [Rules API proposal](https://observatorium.io/docs/proposals/20201019-prometheus-rules-in-observatorium-api.md/) we already added Rules APIs for reading and configuring rules which include both recording and alerting rules.
+
+For example, to save alerts we can already PUT the following object to `/api/metrics/v1/{tenant}/api/v1/rules/raw`
+
+```
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - alert: HighRequestLatency
+        expr: job:request_latency_seconds:mean5m{job="second"} > 0.5
+        for: 10m
+        labels:
+          team: alpha
+```
+
+Such a rule will be saved through Observatorium API, via Rule-Store to object storage for the specified tenant. It can be then listed through a similar endpoint and GET. Such configuration will be then eventually synchronized with tenant’s Rulers using RuleSync (1m poll by default). This is what was agreed so far and is being tested in downstream RHOBS (Red Hat Observability Service) deployment of Observatorium. 
+
+### ConfigSync
+
+The first decision we propose is to move https://github.com/observatorium/rules-objstore to be pure Go library, potentially in new https://github.com/observatorium/config-client repo. Such library would need to support not only saving and retrieving Rules but also Alerting Routing configuration (and potentially silencer, inhibits and other user configuration in the future). Since alerting routing requires secrets we might need to implement various backends in this client (that's why no `objstore` suffix. ). See [chapter about secrets](#managing-user-secrets) for more details. 
```

Comment:
I think I understand what is meant here, but since it is already a 'pure' Go library, maybe wording it a bit differently:

```suggestion
The first decision we propose is to move https://github.com/observatorium/rules-objstore to be a generic configuration library, potentially in new https://github.com/observatorium/config-client repo. Such library would need to support not only saving and retrieving Rules but also Alerting Routing configuration (and potentially silencer, inhibits and other user configuration in the future). Since alerting routing requires secrets we might need to implement various backends in this client (that's why no `objstore` suffix. ). See [chapter about secrets](#managing-user-secrets) for more details. 
```